src/Build/Evaluation/Evaluator.cs

```diff
@@ -1785,7 +1785,14 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                 }
 
                 // Combine SDK path with the "project" relative path
-                sdkResult = _sdkResolverService.ResolveSdk(_submissionId, sdkReference, _evaluationLoggingContext, importElement.Location, solutionPath, projectPath, _interactive, _isRunningInVisualStudio);
+                try
+                {
+                    sdkResult = _sdkResolverService.ResolveSdk(_submissionId, sdkReference, _evaluationLoggingContext, importElement.Location, solutionPath, projectPath, _interactive, _isRunningInVisualStudio);
+                }
+                catch (SdkResolverException e)
+                {
+                    ProjectErrorUtilities.ThrowInvalidProject(importElement.SdkLocation, "SDKResolverCriticalFailure", e.Message);
```

Comment:
Likewise here this seems like it might be able to be better. `e.InnerException.ToString()`?

Reply:
If we log the inner exception's message here we lose the stack trace & resource message that was logged when the resolver failed and end up with:

```
C:\src\issues\6763-sdkresolvers\console>"C:\Users\bevillal\Downloads\dotnet-sdk-6.0.100-preview.7.21379.14-win-x64\dotnet.exe" build
Microsoft (R) Build Engine version 17.0.0-dev-21424-01+2cafba516 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242: SDK Resolver Failure: "'s' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8."

Build FAILED.

C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242: SDK Resolver Failure: "'s' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8."
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.18
```

See the above comment for what it would look like if we use `e.ToString()`


Below is `e.ToString()` with the resolverservice logging its `e.ToString()`. This is just too verbose imo.

```
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242: SDK Resolver Failure: "Microsoft.Build.BackEnd.SdkResolution.SdkResolverException: The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed while attempting to resolve the SDK "Microsoft.NET.Sdk". Exception: "System.Text.Json.JsonReaderException: 's' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8.
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes) in System.Text.Json.dll:token 0x60000de+0x0
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenUntilAfterAllCommentsAreSkipped(Byte marker) in System.Text.Json.dll:token 0x6000315+0x1fc
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextToken(Byte marker) in System.Text.Json.dll:token 0x6000311+0x3c
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenOrRollback(Byte marker) in System.Text.Json.dll:token 0x6000310+0x24
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ReadSingleSegment() in System.Text.Json.dll:token 0x60002fc+0x113
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.Read() in System.Text.Json.dll:token 0x60002e8+0x8
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.Utf8JsonStreamReader.Read() in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x60000ef+0x12f
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinition(WorkloadId id, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000082+0x2ff
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinitions(Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600007c+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String id, String informationalPath, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000079+0x124
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String manifestId, Stream manifestStream, String informationalPath) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000084+0x23
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvider) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600009e+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000099+0x8a
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.CachingWorkloadResolver.Resolve(String sdkReferenceName, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x6000019+0x5b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.WorkloadSdkResolver.Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x600001c+0x50
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.SdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\SdkResolverService.cs:line 118"
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:  ---> System.Text.Json.JsonReaderException: 's' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8.
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes) in System.Text.Json.dll:token 0x60000de+0x0
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenUntilAfterAllCommentsAreSkipped(Byte marker) in System.Text.Json.dll:token 0x6000315+0x1fc
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextToken(Byte marker) in System.Text.Json.dll:token 0x6000311+0x3c
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenOrRollback(Byte marker) in System.Text.Json.dll:token 0x6000310+0x24
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ReadSingleSegment() in System.Text.Json.dll:token 0x60002fc+0x113
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.Read() in System.Text.Json.dll:token 0x60002e8+0x8
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.Utf8JsonStreamReader.Read() in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x60000ef+0x12f
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinition(WorkloadId id, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000082+0x2ff
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinitions(Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600007c+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String id, String informationalPath, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000079+0x124
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String manifestId, Stream manifestStream, String informationalPath) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000084+0x23
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvider) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600009e+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000099+0x8a
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.CachingWorkloadResolver.Resolve(String sdkReferenceName, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x6000019+0x5b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.WorkloadSdkResolver.Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x600001c+0x50
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.SdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\SdkResolverService.cs:line 118
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    --- End of inner exception stack trace ---
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.SdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\SdkResolverService.cs:line 132
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.CachingSdkResolverService.<>n__0(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in Microsoft.Build.dll:token 0x600169e+0x0
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.CachingSdkResolverService.<>c__DisplayClass3_0.<ResolveSdk>b__1() in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\CachingSdkResolverService.cs:line 55
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode) in System.Private.CoreLib.dll:token 0x60013b6+0x1c
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor) in System.Private.CoreLib.dll:token 0x60013b7+0x22
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Lazy`1.CreateValue() in System.Private.CoreLib.dll:token 0x60013bc+0x74
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Lazy`1.get_Value() in System.Private.CoreLib.dll:token 0x60013c2+0xa
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.CachingSdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\CachingSdkResolverService.cs:line 58
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.MainNodeSdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\MainNodeSdkResolverService.cs:line 95
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.Evaluation.Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpressionConditioned(String directoryOfImportingFile, ProjectImportElement importElement, List`1& projects, SdkResult& sdkResult, Boolean throwOnFileNotExistsError) in C:\src\git\msbuild\src\Build\Evaluation\Evaluator.cs:line 1790"
    0 Warning(s)
    1 Error(s)
```

Below is `e.InnerException.ToString()` when resolverservice logs its `e.ToString()`. Here we don't get which sdkresolver failed resolving which sdk.

```
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242: SDK Resolver Failure: "System.Text.Json.JsonReaderException: 's' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8.
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes) in System.Text.Json.dll:token 0x60000de+0x0
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenUntilAfterAllCommentsAreSkipped(Byte marker) in System.Text.Json.dll:token 0x6000315+0x1fc
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextToken(Byte marker) in System.Text.Json.dll:token 0x6000311+0x3c
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenOrRollback(Byte marker) in System.Text.Json.dll:token 0x6000310+0x24
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ReadSingleSegment() in System.Text.Json.dll:token 0x60002fc+0x113
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.Read() in System.Text.Json.dll:token 0x60002e8+0x8
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.Utf8JsonStreamReader.Read() in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x60000ef+0x12f
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinition(WorkloadId id, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000082+0x2ff
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinitions(Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600007c+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String id, String informationalPath, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000079+0x124
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String manifestId, Stream manifestStream, String informationalPath) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000084+0x23
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvider) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600009e+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000099+0x8a
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.CachingWorkloadResolver.Resolve(String sdkReferenceName, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x6000019+0x5b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.WorkloadSdkResolver.Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x600001c+0x50
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.SdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\SdkResolverService.cs:line 118"
    0 Warning(s)
    1 Error(s)
```

And here is evaluator logging `e.Message` while resolverservice logs `e.ToString()`. This one is ideal because we get:
1. Overall MSB4242.
2. Which SDKresolver failed, and which SDK it failed to resolve
3. Inner exception ('s' is an invalid...)
4. Stack trace

```
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242: SDK Resolver Failure: "The SDK resolver "Microsoft.DotNet.MSBuildWorkloadSdkResolver" failed while attempting to resolve the SDK "Microsoft.NET.Sdk". Exception: "System.Text.Json.JsonReaderException: 's' is an invalid start of a property name. Expected a '"'. LineNumber: 14 | BytePositionInLine: 8.
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes) in System.Text.Json.dll:token 0x60000de+0x0
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenUntilAfterAllCommentsAreSkipped(Byte marker) in System.Text.Json.dll:token 0x6000315+0x1fc
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextToken(Byte marker) in System.Text.Json.dll:token 0x6000311+0x3c
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ConsumeNextTokenOrRollback(Byte marker) in System.Text.Json.dll:token 0x6000310+0x24
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.ReadSingleSegment() in System.Text.Json.dll:token 0x60002fc+0x113
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at System.Text.Json.Utf8JsonReader.Read() in System.Text.Json.dll:token 0x60002e8+0x8
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.Utf8JsonStreamReader.Read() in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x60000ef+0x12f
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinition(WorkloadId id, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000082+0x2ff
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadDefinitions(Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600007c+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String id, String informationalPath, Utf8JsonStreamReader& reader) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000079+0x124
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadManifestReader.ReadWorkloadManifest(String manifestId, Stream manifestStream, String informationalPath) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000084+0x23
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.LoadManifestsFromProvider(IWorkloadManifestProvider manifestProvider) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x600009e+0x2b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver.Create(IWorkloadManifestProvider manifestProvider, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadManifestReader.dll:token 0x6000099+0x8a
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.CachingWorkloadResolver.Resolve(String sdkReferenceName, String dotnetRootPath, String sdkVersion) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x6000019+0x5b
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.WorkloadSdkResolver.Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory) in Microsoft.NET.Sdk.WorkloadMSBuildSdkResolver.dll:token 0x600001c+0x50
C:\src\issues\6763-sdkresolvers\console\console.csproj : error MSB4242:    at Microsoft.Build.BackEnd.SdkResolution.SdkResolverService.ResolveSdk(Int32 submissionId, SdkReference sdk, LoggingContext loggingContext, ElementLocation sdkReferenceLocation, String solutionPath, String projectPath, Boolean interactive, Boolean isRunningInVisualStudio) in C:\src\git\msbuild\src\Build\BackEnd\Components\SdkResolution\SdkResolverService.cs:line 118""
    0 Warning(s)
    1 Error(s)
```

Reply:
That last one looks good. A bit chatty but this is always a resolver bug so I prefer to get a bunch of info out. Please comment the Message call so it stands out to future code smellers.